src/Entrees.tsx

```diff
@@ -0,0 +1,114 @@
+import axios from "axios";
+import React, { useState } from "react";
+
+type Entree = {
+	name: { value: string };
+	description: { value: string };
+	price: { value: number };
+} & EventTarget;
+
+const Entrees = () => {
+	const [statusMessage, setStatusMessage] = useState<string>();
+
+	const isValid = (e: React.FormEvent<HTMLFormElement>): boolean => {
+		const target = e.target as Entree;
+		const name = target.name.value;
+		const description = target.description.value;
+
+		if (name.trim() === "") {
+			setStatusMessage("Entree name cannot be blank");
+		} else if (name.length > 100) {
+			setStatusMessage("Entree name cannot be over 100 characters");
+		} else if (description.trim() === "") {
+			setStatusMessage("Entree description cannot be blank");
+		} else if (description.length > 160) {
+			setStatusMessage("description must be less than 160 characters");
+		} else {
+			return true;
+		}
+
+		return false;
+	};
+
+	const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+		e.preventDefault();
+
+		const target = e.target as Entree;
+
+		const name = target.name.value;
+		const description = target.description.value;
+		const price = target.price.value;
+		if (isValid(e)) {
+			axios
+				.post<Entree>("http://localhost:8080/entree", {
+					name,
+					description,
+					price,
+				})
+				.then(() => {
+					setStatusMessage("successfully added a new entree");
+				})
+				.catch(error => {
+					setStatusMessage(error.response.data.message);
+				});
```

Comment:
Minor thing, If the server is not running or is down, there are no errors to tell the user there is a problem with the server. Could we make an if statement for status 500 "The serves are having a problem please come back later"